sklearn/preprocessing/_encoders.py

```diff
@@ -848,3 +852,246 @@ def inverse_transform(self, X):
             X_tr[:, i] = self.categories_[i][labels]
 
         return X_tr
+
+
+class UnaryEncoder(BaseEstimator, TransformerMixin):
+    """Encode ordinal integer features using a unary scheme.
+
+    This encoder transforms each ordinal feature with ``m`` possible values
+    into ``m - 1`` binary features, where the ith feature is active if ``x >
+    i``. The input to this transformer should be a matrix of non-negative
+    integers, denoting the values taken on by the ordinal features.
+
+    This encoding may be needed for feeding ordinal features to many
+    scikit-learn estimators, notably linear models and kernel-based models
+    like SVMs with the standard kernels.
+    This transformation is unlikely to help when using with tree-based models,
+    since those already work on the basis of a particular feature value being
+    less or greater than a threshold, unlike linear and kernel-based models.
+
+    This encoder encodes all of the features. To only encode a subset of the
+    features, use :class:`ColumnTransformer
+    <sklearn.compose.ColumnTransformer>`.
+
+    Read more in the :ref:`User Guide <unary_encoding>`.
+
+    Parameters
+    ----------
+    n_values : 'auto', int or array of ints, optional (default='auto')
```

Comment:
Should this be changed to `categories` to be consistent with other encoders?

Note that in the other encoders `categories` is a list of list, while here it would only be a list of int (the list of the maximum) since the input matrix X is supposed to contain non-negative integers.

Reply:
I think we should turn to ``categories``. Maybe we should support string here.

Reply:
As far as I understand the variables are supposed to be ordinal. How would we represent ordinal features with strings?

Reply:
I think we should support things like
```
enc = preprocessing.UnaryEncoder(categories=[["a", "b", "c"]])
enc.fit_transform([["a"], ["b"], ["c"]])
```
Although users can combine ``UnaryEncoder`` with ``OrdinalEncoder``.

Reply:
Yes I think `OrdinalEncoder` should be used for this since its output is precisely the input that `UnaryEncoder` requires. I'll add an example in the user guide.

Reply:
So you force users to use ``OrdinalEncoder`` first? I'd argue that it's not user-friendly :)
I think ``OrdinalEncoder`` should be similar to ``OneHotEncoder``, but the ``categories`` parameter here is actually similar to the ``n_values`` parameter we previously had in ``OneHotEncoder``.

Reply:
How about we make `categories` being exactly the same as in `OrdinalEncoder`, and rework the implementation of `UnaryEncoder` to first use a `OrdinalEncoder` under the hood, in order to have integer input (as currently required)?

Basically: make `UnaryEncoder` work exactly like `make_pipeline(OrdinalEncoder(categories), CurrentUnaryEncoder())`

Reply:
> Basically: make UnaryEncoder work exactly like make_pipeline(OrdinalEncoder(categories), CurrentUnaryEncoder())

This is exactly what I thought, I'll vote +1.

Reply:
Ok I think this is a good idea too, just want input from @jnothman before reworking this

Reply:
Yes, I'm happy with making this have input the same as OrdinalEncoder. But I also don't require it for this to be merged.